### ISSUE TITLE
Warnings when compiling example 07_no_macros

### ISSUE THREAD

**Issue by rblath** [SanderMertens/flecs#110]
Not sure if it is worth fixing but I get 3 warnings when compiling example 07 on macOS Catalina 10.15.1:

````
~/Projects/GitHub/flecs/examples/07_no_macros $ gcc src/*.c -Iinclude -I../../include -o "`basename $PWD`" ../../build/libflecs_shared.dylib
src/main.c:51:35: warning: incompatible pointer to integer conversion passing 'ecs_type_t' (aka 'const struct ecs_vector_t *') to parameter of type 'ecs_entity_t'
      (aka 'unsigned long long') [-Wint-conversion]
    _ecs_set_ptr(world, MyEntity, TEcsId, sizeof(EcsId), &(EcsId){"MyEntity"});
                                  ^~~~~~
../../include/flecs.h:1147:18: note: passing argument to parameter 'component' here
    ecs_entity_t component,
                 ^
src/main.c:60:35: warning: incompatible pointer to integer conversion passing 'ecs_type_t' (aka 'const struct ecs_vector_t *') to parameter of type 'ecs_entity_t'
      (aka 'unsigned long long') [-Wint-conversion]
    _ecs_set_ptr(world, MyEntity, TPosition, sizeof(Position), &(Position){0, 0});
                                  ^~~~~~~~~
../../include/flecs.h:1147:18: note: passing argument to parameter 'component' here
    ecs_entity_t component,
                 ^
src/main.c:61:35: warning: incompatible pointer to integer conversion passing 'ecs_type_t' (aka 'const struct ecs_vector_t *') to parameter of type 'ecs_entity_t'
      (aka 'unsigned long long') [-Wint-conversion]
    _ecs_set_ptr(world, MyEntity, TVelocity, sizeof(Velocity), &(Velocity){1, 1});
                                  ^~~~~~~~~
../../include/flecs.h:1147:18: note: passing argument to parameter 'component' here
    ecs_entity_t component,
                 ^
3 warnings generated.
````

````
$ gcc --version
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/4.2.1
Apple clang version 11.0.0 (clang-1100.0.33.12)
Target: x86_64-apple-darwin19.0.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
````

Flecs was built with CMake:

```
~/Projects/GitHub/flecs $ mkdir build
~/Projects/GitHub/flecs $ cd build
~/Projects/GitHub/flecs/build $ cmake ..
-- The C compiler identification is AppleClang 11.0.0.11000033
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Configuring done
CMake Warning (dev):
  Policy CMP0042 is not set: MACOSX_RPATH is enabled by default.  Run "cmake
  --help-policy CMP0042" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  MACOSX_RPATH is not specified for the following targets:

   flecs_shared

This warning is for project developers.  Use -Wno-dev to suppress it.

-- Generating done
-- Build files have been written to: /Users/rblath/Projects/GitHub/flecs/build
~/Projects/GitHub/flecs/build $ ll
total 88
-rw-r--r--   1 rblath  staff    12K Dec  5 23:00 CMakeCache.txt
drwxr-xr-x  13 rblath  staff   416B Dec  5 23:00 CMakeFiles
-rw-r--r--   1 rblath  staff    23K Dec  5 23:00 Makefile
-rw-r--r--   1 rblath  staff   3.1K Dec  5 23:00 cmake_install.cmake
~/Projects/GitHub/flecs/build $ make
Scanning dependencies of target flecs_static
[  2%] Building C object CMakeFiles/flecs_static.dir/src/chunked.c.o
[  5%] Building C object CMakeFiles/flecs_static.dir/src/column_system.c.o
[  8%] Building C object CMakeFiles/flecs_static.dir/src/entity.c.o
[ 11%] Building C object CMakeFiles/flecs_static.dir/src/err.c.o
[ 14%] Building C object CMakeFiles/flecs_static.dir/src/map.c.o
[ 17%] Building C object CMakeFiles/flecs_static.dir/src/misc.c.o
[ 20%] Building C object CMakeFiles/flecs_static.dir/src/os_api.c.o
[ 23%] Building C object CMakeFiles/flecs_static.dir/src/parser.c.o
[ 26%] Building C object CMakeFiles/flecs_static.dir/src/stage.c.o
[ 29%] Building C object CMakeFiles/flecs_static.dir/src/stats.c.o
[ 32%] Building C object CMakeFiles/flecs_static.dir/src/system.c.o
[ 35%] Building C object CMakeFiles/flecs_static.dir/src/table.c.o
[ 38%] Building C object CMakeFiles/flecs_static.dir/src/type.c.o
[ 41%] Building C object CMakeFiles/flecs_static.dir/src/vector.c.o
[ 44%] Building C object CMakeFiles/flecs_static.dir/src/worker.c.o
[ 47%] Building C object CMakeFiles/flecs_static.dir/src/world.c.o
[ 50%] Linking C static library libflecs_static.a
[ 50%] Built target flecs_static
Scanning dependencies of target flecs_shared
[ 52%] Building C object CMakeFiles/flecs_shared.dir/src/chunked.c.o
[ 55%] Building C object CMakeFiles/flecs_shared.dir/src/column_system.c.o
[ 58%] Building C object CMakeFiles/flecs_shared.dir/src/entity.c.o
[ 61%] Building C object CMakeFiles/flecs_shared.dir/src/err.c.o
[ 64%] Building C object CMakeFiles/flecs_shared.dir/src/map.c.o
[ 67%] Building C object CMakeFiles/flecs_shared.dir/src/misc.c.o
[ 70%] Building C object CMakeFiles/flecs_shared.dir/src/os_api.c.o
[ 73%] Building C object CMakeFiles/flecs_shared.dir/src/parser.c.o
[ 76%] Building C object CMakeFiles/flecs_shared.dir/src/stage.c.o
[ 79%] Building C object CMakeFiles/flecs_shared.dir/src/stats.c.o
[ 82%] Building C object CMakeFiles/flecs_shared.dir/src/system.c.o
[ 85%] Building C object CMakeFiles/flecs_shared.dir/src/table.c.o
[ 88%] Building C object CMakeFiles/flecs_shared.dir/src/type.c.o
[ 91%] Building C object CMakeFiles/flecs_shared.dir/src/vector.c.o
[ 94%] Building C object CMakeFiles/flecs_shared.dir/src/worker.c.o
[ 97%] Building C object CMakeFiles/flecs_shared.dir/src/world.c.o
[100%] Linking C shared library libflecs_shared.dylib
[100%] Built target flecs_shared
```

By the way: Flecs is an awesome library! Must have taken a lot of work.

**Comment by SanderMertens** [comment]
@rblath thanks for reporting, I'll look into it!

**Comment by SanderMertens** [comment]
Issue has been fixed in this commit: 7ff8a3c7342b67e6b018f9f1e925e9db7d01025b
